packages/amplify_datastore_plugin_interface/test/testData/ModelProvider.dart

```diff
@@ -37,4 +37,24 @@ class ModelProvider implements ModelProviderInterface {
   static final ModelProvider _instance = ModelProvider();
 
   static ModelProvider get instance => _instance;
+
+  ModelType getModelTypeByModelName(String modelName) {
```

Comment:
It seems maybe like you've got some modelgen changes in this PR, that don't need to be here with the Hub changes. To minimize the scope and size of the current PR, I'd cut all this stuff as a separate PR. If its dependent work, we can get this stuff in first.

Reply:
This is needed by the hub change to cast the instances that are included in certain datastore event payloads to their respective model types.